geoh5py/objects/surveys/electromagnetics/airborne_tem.py

```diff
@@ -0,0 +1,388 @@
+        """Data relative_to_bearing"""
+        if "Angles relative to bearing" in self.metadata["EM Dataset"]:
+            return self.metadata["EM Dataset"]["Angles relative to bearing"]
+
+        return None
+
+    @relative_to_bearing.setter
+    def relative_to_bearing(self, value: bool):
+        if not isinstance(value, bool):
+            raise TypeError("Input 'relative_to_bearing' must be one of type 'bool'")
+        self.edit_metadata({"Angles relative to bearing": value})
+
+    @property
+    def roll(self) -> float | uuid.UUID | None:
+        """Roll angle(s) of the transmitter coil."""
+        if "Roll value" in self.metadata["EM Dataset"]:
+            return self.metadata["EM Dataset"]["Roll value"]
+        if "Roll property" in self.metadata["EM Dataset"]:
+            return self.metadata["EM Dataset"]["Roll property"]
+        return None
+
+    @roll.setter
+    def roll(self, value: float | uuid.UUID):
+        if isinstance(value, float):
+            self.edit_metadata({"Roll value": value, "Roll property": None})
+        elif isinstance(value, uuid.UUID):
+            self.edit_metadata({"Roll value": None, "Roll property": value})
+        else:
+            raise TypeError("Input 'roll' must be one of type float or uuid.UUID")
+
+    @property
+    def timing_mark(self) -> float | None:
+        """
+        Timing mark from the beginning of the discrete :attr:`waveform`.
+        Generally used as the reference (time=0.0) for the provided
+        (-) on-time an (+) off-time :attr:`channels`.
+        """
+        if (
+            "Waveform" in self.metadata["EM Dataset"]
+            and "Timing mark" in self.metadata["EM Dataset"]["Waveform"]
+        ):
+            timing_mark = self.metadata["EM Dataset"]["Waveform"]["Timing mark"]
+            return timing_mark
+
+        return None
+
+    @timing_mark.setter
+    def timing_mark(self, timing_mark: float):
+        if not isinstance(timing_mark, float):
+            raise ValueError("Input timing_mark must be a float.")
+
+        self.edit_metadata({"Timing mark": timing_mark})
+
+    @property
+    def waveform(self) -> np.ndarray | None:
+        """
+        :obj:`numpy.array` of :obj:`float`, shape(*, 2): Discrete waveform of the TEM source.
+
+        .. code-block:: python
+
+            waveform = [
+                [time_1, current_1],
+                [time_2, current_2],
+                ...
+            ]
+        """
+        if (
+            "Waveform" in self.metadata["EM Dataset"]
+            and "Discretization" in self.metadata["EM Dataset"]["Waveform"]
+        ):
+            waveform = np.vstack(
+                [
+                    [row["time"], row["current"]]
+                    for row in self.metadata["EM Dataset"]["Waveform"]["Discretization"]
+                ]
+            )
+            return waveform
+        return None
+
+    @waveform.setter
+    def waveform(self, waveform: np.ndarray | None):
```

Comment:
**question:** some setters, as this one allows for `None`,  others, as `timing_mark`, do not.
Should they all allow for `None` and call `del`

Reply:
Done